repos/system_upgrade/common/actors/scansubscriptionmanagerinfo/tests/test_scansubscriptionmanagementinfo.py

```diff
@@ -0,0 +1,349 @@
+import logging
+import os
+from collections import namedtuple
+
+import pytest
+
+from leapp.libraries import stdlib
+from leapp.libraries.actor import scanrhsm
+from leapp.libraries.stdlib import api
+from leapp.libraries.common import mounting
+from leapp.libraries.common import repofileutils
+from leapp.libraries.common import rhsm
+from leapp.libraries.common.testutils import CurrentActorMocked, produce_mocked
+from leapp.models import RepositoryFile, RepositoryData
+
+
+RHSM_LIST_CONSUMED = ('subscription-manager', 'list', '--consumed')
+RHSM_STATUS = ('subscription-manager', 'status')
+RHSM_RELEASE = ('subscription-manager', 'release')
+RHSM_LIST_ENABLED_REPOS = ('subscription-manager', 'repos', '--list-enabled')
+
+RHSM_STATUS_OUTPUT_NONSATELLITE = '''
++-------------------------------------------+
+   System Status Details
++-------------------------------------------+
+Overall Status: Current
+
+System Purpose Status: Matched
+'''
+
+# Not the actual satelitte output (but it should suffice)
+RHSM_STATUS_OUTPUT_SATELLITE = '''
++-------------------------------------------+
+   System Status Details
++-------------------------------------------+
+Overall Status: Current
+
+System Purpose Status: Matched
+
+Content Access Mode is set to Simple Content Access
+'''
+
+RHSMRepositoryEntry = namedtuple('RHSMRepositoryEntry', ('id', 'name', 'url', 'enabled'))  # For clarity purposes
+RHSM_ENABLED_REPOS = [
+    RHSMRepositoryEntry(
+        id='rhel-8-for-x86_64-appstream-rpms',
+        name='Appstream',
+        url='some_url',
+        enabled='1'),
+    RHSMRepositoryEntry(
+        id='satellite-tools-6.6-for-rhel-8-x86_64-rpms',
+        name='Satellite',
+        url='some_url',
+        enabled='1'),
+    RHSMRepositoryEntry(
+        id='rhel-8-for-x86_64-baseos-rpms',
+        name='Base',
+        url='some_url',
+        enabled='1')
+]
+
+
+class MockedCommandRunner(object):
+    """Mock for the api.run that allows mapping of called commands to desired output."""
+    def __init__(self):
+        self.mocked_command_calls = dict()  # A map from called commands to their mocked output
+        self.calls_performed_counter = 0
+
+    def add_mocked_command_with_stdout(self, command, stdout):
+        self.mocked_command_calls[command] = {'stdout': stdout}
+
+    def __call__(self, command, **kwargs):
+        # Ignore kwargs
+        default_result = {'stdout': ''}
+        self.calls_performed_counter += 1
+        return self.mocked_command_calls.get(tuple(command), default_result)
```

Comment:
Take baseically the current file and extend the `common/libraries/tests/test_rhsm.py` (see the `IsolatedActionsMocked`).

Reply:
Suggested changes were implemented in the most recent commits.